crates/holochain_types/src/web_app/web_app_manifest.rs

```diff
@@ -0,0 +1,61 @@
+#![warn(missing_docs)]
+
+//! Defines the hApp Manifest YAML format, including validation.
+
+use mr_bundle::{Location, Manifest};
+use std::path::PathBuf;
+
+mod current;
+pub(crate) mod web_app_manifest_v1;
+
+pub use current::*;
+
+use web_app_manifest_v1::WebAppManifestV1;
+
+/// Container struct which uses the `manifest_version` field to determine
+/// which manifest version to deserialize to.
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize, derive_more::From)]
+#[serde(tag = "manifest_version")]
+#[allow(missing_docs)]
+pub enum WebAppManifest {
+    #[serde(rename = "1")]
+    V1(WebAppManifestV1),
+}
+
+impl Manifest for WebAppManifest {
+    fn locations(&self) -> Vec<Location> {
+        match self {
+            WebAppManifest::V1(m) => vec![m.ui.location.clone(), m.happ_manifest.location.clone()],
+        }
+    }
+
+    fn path() -> PathBuf {
+        "web-happ.yaml".into()
+    }
+
+    fn bundle_extension() -> &'static str {
+        "web-happ"
```

Comment:
@zippy what extension should the final bundle have?

Reply:
For now lets use .webhapp

Reply:
```suggestion
        "webhapp"
```